File: src/main/php/VeryEasy/TemperatureConverter.php
<?php

declare(strict_types=1);

namespace Src\VeryEasy;

class TemperatureConverter
{
    public function convertTemperature(float $celsius): array
    {
        return [
            $this->toKelvin($celsius),
            $this->toFahrenheit($celsius),
        ];
    }

    public function toCelsius(float $fahrenheit): float
    {
        return ($fahrenheit - 32) * 5.0 / 9.0;
    }

    public function toFahrenheit(float $celsius): float
    {
        return ($celsius * 1.8) + 32;
    }

    public function toKelvin(float $celsius): float
    {
        return $celsius + 273.15;
    }

    public function transformTemperatureConverter(string $deg): string
    {
        $temp = preg_replace('/[^0-9\.\-]/', '', $deg);
        if ($temp === '' || ! is_numeric($temp)) {
            return 'Error';
        }
        $temperature = (float) $temp;

        if (str_ends_with($deg, '°F')) {
            return sprintf('%.0f°C', self::toCelsius($temperature));
        }

        if (str_ends_with($deg, '°C')) {
            return sprintf('%.0f°F', self::toFahrenheit($temperature));
        }

        return 'Error';
    }

    public function solve(string $temperature): string
    {
        return $this->transformTemperatureConverter($temperature);
    }
}

File: src/very_easy/temperature_converter.rs
pub fn tolerance() -> f64 {
	1e-5
}
pub fn to_fahrenheit(celsius: f64) -> f64 {
	(celsius * 1.8) + 32.0
}
pub fn to_kelvin(celsius: f64) -> f64 {
	celsius + 273.15
}
pub fn to_celsius(fahrenheit: f64) -> f64 {
	(fahrenheit - 32.0) *( 5.0 / 9.0)
}
pub fn temperature_converter(deg: String) -> String {
  let temperature = deg.chars().filter(|&t| t.is_ascii_digit() || t == '.' || t == '-').collect::<String>().parse::<f64>().unwrap_or(0.0);
	if deg.ends_with("°C") {
		return format!("{}°F", to_fahrenheit(temperature).round() as i32);
	}
	else if deg.ends_with("°F") {
		return format!("{}°C", to_celsius(temperature).round() as i32);
		}else {
			return "Error".to_string();
		}

}

pub fn convert_temperature(celsius: f64) -> Vec<f64> {
	Vec::from([
		to_kelvin(celsius),
		to_fahrenheit(celsius),
	])
}
